errors_test.go

```diff
@@ -0,0 +1,38 @@
+package cas
+
+import "testing"
+
+// expected invalid token
+// {
+//	"error":{
+//		"code":1000,
+//		"message":"Invalid token"
+//	}
+// }
+func TestRequestError_Error(t *testing.T) {
+	client := New("abc", "xyz")
+
+	_, err := client.Movie("12313213")
```

Comment:
It actually makes an http request to the server. I'm not sure if we need any functional tests for this library.

How about just call Error() from an instantiated RequestError? Since RequestError barely have any logic in it, we can possibly remove the test.

Reply:
The test is just want to ensure that to the code and the message response is matched to the TwitCasting API v2 expected

Reply:
but ok we can remove it

Reply:
client.Movie() makes a real http request, and it depends on the twitcasting.tv server's condition. We'd like to make the test with less external dependency. It fails if it's offline:

```
go test
--- FAIL: TestRequestError_Error (0.00s)
    errors_test.go:20: <nil> is not RequestError
FAIL
```

Reply:
How about either:

- Call Error() from an instantiated RequestError
- Remove the entire test (TestRequestError_Error)

For smoke test, I think actually examples/current_live/current_live.go would be enough as it uses RequestError.

Reply:
> 

oh so let's just use nil catching for handling offline error ?

Reply:
so just to remove the error_test.go is enough?

Reply:
> client.Movie() makes a real http request, and it depends on the twitcasting.tv server's condition. We'd like to make the test with less external dependency. It fails if it's offline:
> 
> ```
> go test
> --- FAIL: TestRequestError_Error (0.00s)
>     errors_test.go:20: <nil> is not RequestError
> FAIL
> ```

may I know what is the real error, does it become the HTTP status error that I make via `fmt.Errorf` ?

Reply:
or how about this?
```go
func TestRequestError_Error(t *testing.T) {
	client := New("abc", "xyz")

	_, err := client.Movie("12313213")

	if requestError, ok := err.(*RequestError); ok {
		t.Logf("RequestError: %+v", requestError)
		return
	} else {
		t.Logf("OtherError: %+v", err)
	}
	
}
```

Reply:
fine, let's remove it

Reply:
Did it make sense why we don't want to call `client.Movie()` in the test?